packages/core/src/list-transactions.ts

```diff
@@ -0,0 +1,144 @@
+      deploymentDir,
+    });
+  }
+
+  const executionOrder = await getExecutionOrder(deploymentLoader);
+  const transactions: ListTransactionsResult = [];
+
+  for (const futureId of executionOrder) {
+    const exState = deploymentState.executionStates[futureId];
+
+    if (!doesSendTransactions(exState)) {
+      continue;
+    }
+
+    for (const networkInteraction of exState.networkInteractions) {
+      assertIgnitionInvariant(
+        networkInteraction.type === "ONCHAIN_INTERACTION",
+        "Expected network interaction to be an onchain interaction"
+      );
+
+      for (const transaction of networkInteraction.transactions) {
+        switch (exState.type) {
+          case ExecutionSateType.DEPLOYMENT_EXECUTION_STATE: {
+            transactions.push({
+              type: exState.type,
+              from: exState.from,
+              txHash: transaction.hash,
+              status: getTransactionStatus(transaction),
+              name: exState.contractName,
+              address:
+                transaction.receipt?.status === TransactionReceiptStatus.SUCCESS
+                  ? exState.result?.type === ExecutionResultType.SUCCESS
+                    ? exState.result.address
+                    : undefined
+                  : undefined,
+              params: exState.constructorArgs,
+              value: networkInteraction.value,
+            });
+
+            break;
+          }
+          case ExecutionSateType.CALL_EXECUTION_STATE: {
+            transactions.push({
+              type: exState.type,
+              from: exState.from,
+              txHash: transaction.hash,
+              status: getTransactionStatus(transaction),
+              name: exState.functionName,
+              to: networkInteraction.to,
+              params: exState.args,
+              value: networkInteraction.value,
+            });
+
+            break;
+          }
+          case ExecutionSateType.SEND_DATA_EXECUTION_STATE: {
+            transactions.push({
+              type: exState.type,
+              from: exState.from,
+              txHash: transaction.hash,
+              status: getTransactionStatus(transaction),
+              to: networkInteraction.to,
+              value: networkInteraction.value,
+            });
+
+            break;
+          }
+        }
+      }
+    }
+  }
+
+  return transactions;
+}
+
+function doesSendTransactions(
+  exState: ExecutionState
+): exState is
+  | DeploymentExecutionState
+  | CallExecutionState
+  | SendDataExecutionState {
+  return (
+    exState.type === ExecutionSateType.DEPLOYMENT_EXECUTION_STATE ||
+    exState.type === ExecutionSateType.CALL_EXECUTION_STATE ||
+    exState.type === ExecutionSateType.SEND_DATA_EXECUTION_STATE
+  );
+}
+
+function getTransactionStatus(transaction: Transaction): TransactionStatus {
+  const status =
+    transaction.receipt === undefined
+      ? TransactionStatus.DROPPED
```

Comment:
Can't this also mean pending? I guess dropped would be if that networkInteraction has another transaction after this one.